### PR TITLE
test enum macro consistency

### PR DESCRIPTION
**Pull request by ajax16384**
allows to check for LINE( presense in x86 feature and microarch arrays

### CODE REVIEW

**Comment by ajax16384** [comment]
might be extended to aarch64,arm,mips,ppc feature as well

**Comment by toor1245** [comment]
> might be extended to aarch64,arm,mips,ppc feature as well

I think it is good validation and probably yes we should add this check for other archs.
@gchatelet, any objections?

File: test/cpuinfo_x86_test.cc [hunk 1]
```diff
@@ -118,6 +118,30 @@ class CpuidX86Test : public ::testing::Test {
   }
 };
 
+TEST_F(CpuidX86Test, X86MicroarchitectureEnum) {
+   const char *last_name = GetX86MicroarchitectureName(X86_MICROARCHITECTURE_LAST_);
+   EXPECT_STREQ(last_name, "unknown microarchitecture");
+   for (int i = static_cast<int>(X86_UNKNOWN); i != static_cast<int>(X86_MICROARCHITECTURE_LAST_); ++i) {
+      const auto micro = static_cast<X86Microarchitecture>(i);
+      const char *name = GetX86MicroarchitectureName(micro);
+      EXPECT_FALSE(name == nullptr);
```
Comment (gchatelet) [inline test/cpuinfo_x86_test.cc:127]: This one should be an `ASSERT_FALSE` (next line is UB if this one fails). Same in the function below.

**Comment by gchatelet** [comment]
Yep that's a great idea. @ajax16384 can you add similar tests to other architectures as well?
Thx a lot for the PR.

File: test/cpuinfo_x86_test.cc (hunk 1 above)
Comment (ajax16384) [reply test/cpuinfo_x86_test.cc]: ok

**Comment by ajax16384** [comment]
> Yep that's a great idea. @ajax16384 can you add similar tests to other architectures as well? Thx a lot for the PR.

added rest of arch features enum tests

**Pull request merged**